sda/processor_test.go

```diff
@@ -10,18 +10,78 @@ import (
 	"github.com/dedis/cothority/log"
 	"github.com/dedis/cothority/network"
 	"github.com/dedis/crypto/config"
+	"github.com/stretchr/testify/assert"
 )
 
+type basicProcessor struct {
+	msgChan chan network.Packet
+}
+
+func (bp *basicProcessor) Process(msg *network.Packet) {
+	bp.msgChan <- *msg
+}
+
+type basicMessage struct {
+	Value int
+}
+
+var basicMessageType network.MessageTypeID
+
+func TestBlockingDispatcher(t *testing.T) {
+	defer log.AfterTest(t)
+
+	dispatcher := NewBlockingDispatcher()
+	processor := &basicProcessor{make(chan network.Packet, 1)}
+
+	dispatcher.RegisterProcessor(processor, basicMessageType)
+	dispatcher.Dispatch(&network.Packet{
+		Msg:     basicMessage{10},
+		MsgType: basicMessageType})
+
+	select {
+	case m := <-processor.msgChan:
+		msg, ok := m.Msg.(basicMessage)
+		assert.True(t, ok)
+		assert.Equal(t, msg.Value, 10)
+	default:
```

Comment:
Channel is a buffered one and it's a blocking dispatcher so before the `select`, the message is already into the channel. https://play.golang.org/p/aDtiNV4wms